### PR TITLE
Update contribution documentation

### PR DESCRIPTION
**Pull request by wichert**
Update CONTRIBUTIONS.md:

- add instructions on setting up a local gopath. This allows VS Code to find generated files
- document pre-commit

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -28,3 +36,44 @@ To update the go dependencies for bazel using `go.mod`:
...
+```
+
+## Visual Studio Code
+
+In order for [VS Code](https://code.visualstudio.com) to find generated files
+you will need to create a virtual GOPATH and point VS Code to it. First run this
+command:
+
+```shell
+bazel :gopath
```
Comment (aignas) [inline CONTRIBUTING.md:68]:
```bash
$ bazel :gopath
Command ':gopath' not found. Try 'bazel help'.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -8,8 +15,9 @@ managing go dependencies, otherwise, add it manually to your shell.
 
 ## Bazel
 
-We use [bazel](https://bazel.build) for building and running all tests. The
-following bazel commands will be useful for go-zserio:
+You will need to use  [bazel](http://bazel.build) if you want to build go-zserio
+or run its tests. We use bazel to install require build tools, generate test data,
```
Comment (aignas) [inline CONTRIBUTING.md:19]:
```suggestion
or run its tests. We use bazel to install required build tools, generate test data,
```

Comment (wichert) [reply CONTRIBUTING.md]: Fixed!

File: CONTRIBUTING.md (hunk 1 above)
Comment (wichert) [reply CONTRIBUTING.md:68]: I have no idea where my change to `BUILD.bazel` to add that disappeared to.. fixed now.

Comment (wichert) [reply CONTRIBUTING.md]: Fixed!

File: BUILD.bazel [hunk 3]
```diff
@@ -30,3 +30,11 @@ go_library(
     importpath = "github.com/woven-planet/go-zserio",
     visibility = ["//visibility:private"],
 )
+
+go_path(
```
[the remark this answers is not in the document]
Comment (wichert) [reply BUILD.bazel:33]: https://github.com/bazelbuild/rules_go/wiki/Editor-setup seems relevant here. I'll give that a try tomorrow.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -28,3 +36,69 @@ To update the go dependencies for bazel using `go.mod`:
...
+In order for [VS Code](https://code.visualstudio.com) you will need to configure
+it to use a bazel-aware _gopackagesdriver_ with the gopls language server. To do
+that use the following for `.vscode/settings.json`, replacing
+`/Users/wichert/Code/go-zserio` with the correct location on your machine.
+
+```json
+{
+  "go.goroot": "/Users/wichert/Code/go-zserio/bazel-go-zserio/external/go_sdk",
+  "go.toolsEnvVars": {
+    "GOPACKAGESDRIVER": "/Users/wichert/Code/go-zserio/bin/gopackagesdriver.sh"
```
Comment (aignas) [inline CONTRIBUTING.md:72]: Why don't we use the `{workspaceFolder}` variable in VSCode as mentioned in https://github.com/bazelbuild/rules_go/wiki/Editor-setup#visual-studio-code? That way people can just copy paste the config and not worry. :)

**Review by aignas — APPROVED** [review]
LGTM.

File: CONTRIBUTING.md (hunk 4 above)
Comment (wichert) [reply CONTRIBUTING.md:72]: Because it doesn't actually work: microsoft/vscode#2809

**Pull request merged**